pkg/common/utils/utils.go

```diff
@@ -213,6 +213,51 @@ func GetApplicationIDFromPod(pod *v1.Pod) string {
 	return GenerateApplicationID(pod.Namespace, conf.GetSchedulerConf().GenerateUniqueAppIds, string(pod.UID))
 }
 
+func CheckAppIdInPod(pod *v1.Pod) error {
+	if err := ValidatePodLabelAnnotation(pod, constants.AppIdLabelKeys, constants.AppIdAnnotationKeys); err != nil {
+		return fmt.Errorf("pod has inconsistent application ID in labels and annotations. %w", err)
+	}
+	return nil
+}
+
+func CheckQueueNameInPod(pod *v1.Pod) error {
+	if err := ValidatePodLabelAnnotation(pod, constants.QueueLabelKeys, constants.QueueAnnotationKeys); err != nil {
+		return fmt.Errorf("pod has inconsistent queue name in labels and annotations. %w", err)
+	}
+	return nil
```

Comment:
I think we need the error messages to indicate *which* metadata sources were inconsistent. For example:

"pod has inconsistent application metadata: label applicationID: foo != annotation yunikorn.org/app-id: bar"

Dynamically populate this based on which elements were in conflict.